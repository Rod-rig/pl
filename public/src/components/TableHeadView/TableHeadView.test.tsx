import Table from '@material-ui/core/Table';
import {createMount} from '@material-ui/core/test-utils';
import * as React from 'react';
import {TableHeadView} from './';

const sort = ['position', 'team', 'points'];

describe('TableHeadView', () => {
  it('should exist', () => {
    expect(<TableHeadView chars={sort}/>).toBeDefined();
  });

  it('should render correctly with default props', () => {
    const tree = createMount()(
      <Table>
        <TableHeadView chars={sort}/>
      </Table>);
    expect(tree.html()).toMatchSnapshot();
  });

  it('should render correctly with different props', () => {
    const tree = createMount()(
      <Table>
        <TableHeadView order='desc' sort={sort[1]} chars={sort}/>
      </Table>);
    expect(tree.html()).toMatchSnapshot();
  });

  it('should render correctly with another props', () => {
    const tree = createMount()(
      <Table>
        <TableHeadView order='asc' sort={sort[sort.length - 1]} chars={sort}/>
      </Table>);
    expect(tree.html()).toMatchSnapshot();
  });

  // it('should render correctly with default props', () => {
  //   const tree = renderer
  //     .create(<TableHeadView chars={sort}/>)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
  //
  // it('should render correctly with different props', () => {
  //   const tree = renderer
  //     .create(
  //       <TableHeadView
  //         order='desc'
  //         sort={sort[1]}
  //         chars={sort}
  //       />)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
  //
  // it('should render correctly with another props', () => {
  //   const tree = renderer
  //     .create(
  //       <TableHeadView
  //         order='asc'
  //         sort={sort[sort.length - 1]}
  //         chars={sort}
  //       />)
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
