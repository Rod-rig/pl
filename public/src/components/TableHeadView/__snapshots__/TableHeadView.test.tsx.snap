// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeadView should render correctly with another props 1`] = `"<table class=\\"MuiTable-root-1\\"><thead class=\\"MuiTableHead-root-11\\"><tr class=\\"MuiTableRow-root-12 MuiTableRow-head-15\\"><th class=\\"MuiTableCell-root-17 MuiTableCell-head-18 MuiTableCell-paddingCheckbox-23 Component-cell-2 \\" aria-sort=\\"descending\\" scope=\\"col\\"><span tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiTableSortLabel-root-32\\" role=\\"button\\" title=\\"Sort\\"><svg class=\\"MuiSvgIcon-root-40 MuiTableSortLabel-icon-34 MuiTableSortLabel-iconDirectionAsc-36\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root-17 MuiTableCell-head-18 MuiTableCell-paddingCheckbox-23 Component-cell-2 Component-team-9\\" aria-sort=\\"descending\\" scope=\\"col\\"><span tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiTableSortLabel-root-32\\" role=\\"button\\" title=\\"Sort\\">team<svg class=\\"MuiSvgIcon-root-40 MuiTableSortLabel-icon-34 MuiTableSortLabel-iconDirectionAsc-36\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root-17 MuiTableCell-head-18 MuiTableCell-paddingCheckbox-23 Component-cell-2 \\" aria-sort=\\"descending\\" scope=\\"col\\"><span tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiTableSortLabel-root-32 MuiTableSortLabel-active-33\\" role=\\"button\\" title=\\"Sort\\">points<svg class=\\"MuiSvgIcon-root-40 MuiTableSortLabel-icon-34 MuiTableSortLabel-iconDirectionAsc-36\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th></tr></thead></table>"`;

exports[`TableHeadView should render correctly with default props 1`] = `"<table class=\\"MuiTable-root-1\\"><thead class=\\"MuiTableHead-root-11\\"><tr class=\\"MuiTableRow-root-12 MuiTableRow-head-15\\"><th class=\\"MuiTableCell-root-17 MuiTableCell-head-18 MuiTableCell-paddingCheckbox-23 Component-cell-2 \\" aria-sort=\\"ascending\\" scope=\\"col\\"><span tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiTableSortLabel-root-32\\" role=\\"button\\" title=\\"Sort\\"><svg class=\\"MuiSvgIcon-root-40 MuiTableSortLabel-icon-34 MuiTableSortLabel-iconDirectionDesc-35\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root-17 MuiTableCell-head-18 MuiTableCell-paddingCheckbox-23 Component-cell-2 Component-team-9\\" aria-sort=\\"ascending\\" scope=\\"col\\"><span tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiTableSortLabel-root-32\\" role=\\"button\\" title=\\"Sort\\">team<svg class=\\"MuiSvgIcon-root-40 MuiTableSortLabel-icon-34 MuiTableSortLabel-iconDirectionDesc-35\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root-17 MuiTableCell-head-18 MuiTableCell-paddingCheckbox-23 Component-cell-2 \\" aria-sort=\\"ascending\\" scope=\\"col\\"><span tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiTableSortLabel-root-32\\" role=\\"button\\" title=\\"Sort\\">points<svg class=\\"MuiSvgIcon-root-40 MuiTableSortLabel-icon-34 MuiTableSortLabel-iconDirectionDesc-35\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th></tr></thead></table>"`;

exports[`TableHeadView should render correctly with different props 1`] = `"<table class=\\"MuiTable-root-1\\"><thead class=\\"MuiTableHead-root-11\\"><tr class=\\"MuiTableRow-root-12 MuiTableRow-head-15\\"><th class=\\"MuiTableCell-root-17 MuiTableCell-head-18 MuiTableCell-paddingCheckbox-23 Component-cell-2 \\" aria-sort=\\"ascending\\" scope=\\"col\\"><span tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiTableSortLabel-root-32\\" role=\\"button\\" title=\\"Sort\\"><svg class=\\"MuiSvgIcon-root-40 MuiTableSortLabel-icon-34 MuiTableSortLabel-iconDirectionDesc-35\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root-17 MuiTableCell-head-18 MuiTableCell-paddingCheckbox-23 Component-cell-2 Component-team-9\\" aria-sort=\\"ascending\\" scope=\\"col\\"><span tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiTableSortLabel-root-32 MuiTableSortLabel-active-33\\" role=\\"button\\" title=\\"Sort\\">team<svg class=\\"MuiSvgIcon-root-40 MuiTableSortLabel-icon-34 MuiTableSortLabel-iconDirectionDesc-35\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root-17 MuiTableCell-head-18 MuiTableCell-paddingCheckbox-23 Component-cell-2 \\" aria-sort=\\"ascending\\" scope=\\"col\\"><span tabindex=\\"0\\" class=\\"MuiButtonBase-root-37 MuiTableSortLabel-root-32\\" role=\\"button\\" title=\\"Sort\\">points<svg class=\\"MuiSvgIcon-root-40 MuiTableSortLabel-icon-34 MuiTableSortLabel-iconDirectionDesc-35\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th></tr></thead></table>"`;
